core/vm/environment.go

```diff
@@ -35,6 +37,7 @@ type (
 
 // Context provides the EVM with auxilary information. Once provided it shouldn't be modified.
 type Context struct {
+	context.Context
```

Comment:
If you can run without context, I'd suggest adding a `Cancel` method on environment that flips the atomic stop value.